snforge_std/src/byte_array.cairo

```diff
@@ -1,7 +1,27 @@
+use core::option::OptionTrait;
+use core::byte_array::BYTE_ARRAY_MAGIC;
+
 fn byte_array_as_felt_array(self: @ByteArray) -> Array<felt252> {
     let mut serialized = array![];
 
     self.serialize(ref serialized);
 
     serialized
 }
+
+/// This function is meant to transform a serialized output from a contract call into a `ByteArray`.
+/// `x` - Span of `felt252`s returned from a contract call (panic data)
+/// Returns the parsed `ByteArray`, or an `Err` if the parsing failed.
+fn try_deserialize_bytearray_error(x: Span<felt252>) -> Result<ByteArray, ByteArray> {
+    if x.len() > 0 && *x.at(0) == BYTE_ARRAY_MAGIC {
+        let mut x_span = x.slice(1, x.len() - 1);
+        let deserialization = Serde::<ByteArray>::deserialize(ref x_span);
+            if deserialization.is_none() {
+                return Result::Err("Malformed input provided");
+            }
+            
+            return Result::Ok(deserialization.unwrap());    
+        }
+    
+    Result::Err("Input is not a ByteArray-formatted error")
```

Comment:
Custom `enum` for error type would be better i think. It is more discoverable and easier to document and `match` on. it would also be consistent with other functions

Reply:
Overkill imho, it's not something that you want to catch really. We just want to provide some context here.